### PR TITLE
Issue 5499 - wrong unit from SC shown in typography settings

### LINKED ISSUE #5499: Responsive mechanism for Typography global in SC
### Description

1) Looks like only for Font-size and Line-height.
Should work as paragraph or like typography responsive mechanism for simple sidebar settings
image
2) 
Units dont work correct 
image

### Step-by-step reproduction instructions

1) create custom SC
2) open Typography global
3) change Font-size and Line-height for some of H-tags
4) check inheriting for responsive

### Screenshots, screen recording, code snippet

1) https://jam.dev/c/aac4fdfe-d0cf-4e0b-b3b5-d92941fc7579

### URL and/or Code editor

_No response_

### How to fix it

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by myroslavve**
# Description
First point from issue is a correct behaviour because style card has default values on those breakpoints, so if we want it to have same values on all breakpoints the default style card should be changed.

Fixed second point for typography, also should work for other units in style cards as well.

Fixes #5499 

# How Has This Been Tested?
Checked that units from SC are shown correctly in block settings.

# Test checklist

**_ Front/Back Testing _**

-   [ ] Check that units from SC are shown correctly in block settings.

**_ Pre-Code Testing _**

-   [ ] Check that units from SC are shown correctly in block settings.

# Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] My changes generate no new warnings/errors
-   [x] I have added/updated tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved handling of typography properties such as font size, line height, letter spacing, text indent, word spacing, and bottom gap in the TypographyControl component.
  - Removed default values for several typography-related units to ensure better customization.

- **Refactor**
  - Reorganized and optimized import statements across various modules for better code maintainability.
  - Simplified attribute handling logic in multiple components and utility functions.

- **Tests**
  - Updated test cases to reflect changes in responsive attributes and typography controls, ensuring accurate initialization and saving of units.

### CODE REVIEW

**Review by Olekrut — CHANGES_REQUESTED** [review]
2nd point about units is fixed
https://jam.dev/c/30d01fb7-c2a6-4796-be0b-90a48f4bc750
1st point about responsive mechanism is not fixed
compare sidebar setting and SC
sidebar is ok, SC is not ok
SC shoulr work as sidebar responsive mechanism

**Comment by myroslavve** [comment]
> 2nd point about units is fixed https://jam.dev/c/30d01fb7-c2a6-4796-be0b-90a48f4bc750 1st point about responsive mechanism is not fixed compare sidebar setting and SC sidebar is ok, SC is not ok SC shoulr work as sidebar responsive mechanism

Enabled responsive mechanism for style cards, but for the heading it is a correct behaviour because default maxi style card has values for each breakpoint there by default, so they shouldn't inherit from higher breakpoint, while block doesn't have those defaults
image

**Review by Olekrut — CHANGES_REQUESTED** [review]
1) https://jam.dev/c/2addf682-1ed4-432c-9744-e8736461c412
Not sure that it is related but check it pls. Sidebar settings > typography > 
set font-size 80 and unit %, then change unit for line-height to %, value will be changed automatically
2) https://jam.dev/c/f30f865d-fa09-4bfa-832a-d5ba7f55077c
XL - base, if change units for font-size and line-height, it will be inherited to XXL, if SC should not inherit then it is as well

**Comment by myroslavve** [comment]
> 1. https://jam.dev/c/2addf682-1ed4-432c-9744-e8736461c412
>    Not sure that it is related but check it pls. Sidebar settings > typography >
>    set font-size 80 and unit %, then change unit for line-height to %, value will be changed automatically
Fixed.

> 2. https://jam.dev/c/f30f865d-fa09-4bfa-832a-d5ba7f55077c
>    XL - base, if change units for font-size and line-height, it will be inherited to XXL, if SC should not inherit then it is as well

It is not that SC should not inherit, but in the headings example, each breakpoint has a different value, so they should not then be changed with higher breakpoint according to how responsive works, if you put same values on each breakpoint there, they should then inherit from higher breakpoint.

**Review by Olekrut — APPROVED** [review]

**Review by elzadj — CHANGES_REQUESTED** [review]
Everything aside of the commented out code looks good :+1:

File: e2e-tests/components/typography-control/index.spec.js [hunk 1]
```diff
@@ -442,6 +464,50 @@ describe('TypographyControl', () => {
 		expect(await getBlockStyle(page)).toMatchSnapshot();
 	});
 
+	// it.only('Should save units along the values', async () => {
```
[the remark this answers is not in the document]
Comment (myroslavve) [reply e2e-tests/components/typography-control/index.spec.js]: oops, shouldn't be there

**Review by elzadj — APPROVED** [review]

**Pull request merged**